v2/tools/generator/internal/config/property_configuration.go

```diff
@@ -28,12 +28,21 @@ type PropertyConfiguration struct {
 	armReference                     configurable[bool]   // Specify whether this property is an ARM reference
 	isSecret                         configurable[bool]   // Specify whether this property is a secret
 	isResourceLifecycleOwnedByParent configurable[bool]
-	exportAsConfigMapPropertyName    configurable[string] // The name of the exportAsConfigMap property.
+	exportAsConfigMapPropertyName    configurable[string]             // The name of the exportAsConfigMap property.
+	inputConfigMapMode               configurable[InputConfigMapMode] // The config map mode
 }
 
+type InputConfigMapMode string
+
+const (
+	InputConfigMapModeOptional = "optional"
+	InputConfigMapModeRequired = "required"
+)
+
 const (
 	armReferenceTag                     = "$armReference"                     // Bool specifying whether a property is an ARM reference
 	isSecretTag                         = "$isSecret"                         // Bool specifying whether a property contains a secret
+	inputConfigMapModeTag               = "$inputConfigMapMode"               // string specifying the inputConfigMapMode mode
```

Comment:
Do we need `input` as a part of this? How abuut just `$configMapMode`?

Reply:
I wanted to draw a distinction from the `ObjectModelConfiguration.AzureGeneratedConfigs` and `PropertyConfiguration.ExportAsConfigMapPropertyName`.

I'm worried that `ConfigMapMode` makes it sound like the mode for `ExportAsConfigMapPropertyName` and I want to make sure it's understood those are two different configurations. One is for import and one is for export.

Thoughts?

Reply:
I see what you're trying to do. "Input" seems like a word that's ripe for misinterpretation though. 

Since it's contrasting with `Export`, how about `Import`? #unsure

Reply:
Will change this to `importConfigMapMode`. Agree that contrasts better with the existing `export`. I've also moved the definition of this property down next to the `export` one so that it's clearer that they're related.